src/components/InlineNotification/InlineNotification.js

```diff
@@ -0,0 +1,100 @@
+import PropTypes from 'prop-types';
+import styled, { css } from 'react-emotion';
+import { sizes, colorNames } from '../../styles/constants';
+
+const { SUCCESS, DANGER, WARNING } = colorNames;
+const { MEGA, GIGA } = sizes;
+
+const baseStyles = css`
+  label: inline-notification;
+`;
+
+const marginStyles = ({ noMargin }) =>
+  noMargin &&
+  css`
+    label: text--no-margin;
+    margin-bottom: 0;
+  `;
+
+const createLeftBorder = ({ theme, size, type }) => {
+  const colors = {
+    [DANGER]: theme.colors.r500,
+    [SUCCESS]: theme.colors.g700,
+    [WARNING]: theme.colors.y700
+  };
+
+  return `
+    label: inline-notification--${type};
+    color: ${colors[type]};
+    position: relative;
+    margin-bottom: ${theme.spacings.mega};
+    &:before {
+      display: inline-block;
+      border-top-right-radius: ${theme.borderRadius[size]};
+      border-bottom-right-radius: ${theme.borderRadius[size]};
+      content: '';
+      position: absolute;
+      left: -${theme.spacings[size]};
+      top: 0;
+      height: 100%;
+      background-color: ${colors[type]};
+      width: 3px;
+    }
+  `;
+};
+
+const successStyles = ({ theme, size, type }) =>
```

Comment:
I know writing it like this makes stuff a bit cleaner because you need less logic for figuring out which style to render.

But I think in other components we've put all the logic for one modifier into a single style function. Essentially just checking if `type` is truthy. PropTypes  should take care of verifying the value and the function would early return a falsy value, if `type` is invalid.

If you still want to keep these styles separate, I guess that's OK. I would move the `css` call into `createLeftBorder` though and maybe not destructure anything here? You could also just check `props.type` and forward props to `createLeftBorder`.

Reply:
Sure, I'm pretty much just copying what I'm seeing elsewhere in the code -- not because of any opinions on the finer points of writing emotion 😂

That said, I would treat this as my last priority / change it later given the rest of the things I need to tackle before my meeting block...